Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/InputActionsEditorWindow.cs

```diff
@@ -210,6 +210,14 @@ private void DirtyInputActionsEditorWindow(InputActionsEditorState newState)
         private bool HasAssetChanged(SerializedObject serializedAsset)
         {
             var asset = (InputActionAsset)serializedAsset.targetObject;
+
+            // Checks if the asset being edited is a new asset that was never saved before.
+            // If it is, there's nothing to save.
+            // At the moment, an asset only has the default asset layout content on disk when it is first created.
+            // So in this case we cannot go through the normal path and compare what's on disk with what has been serialized.
+            if (InputActionAsset.HasDefaultJsonLayout(m_AssetJson) && asset.IsEmpty())
```

Comment:
Isn't this specific to this editor? I would recommend moving the logic to the associated utils class or making it an editor-only extension for the asset?

Reply:
Thanks. Remove the method and just have it inline as I don't know if it needs to be used elsewhere at the moment.
Checking if the asset is empty is not editor dependent.